### PR TITLE
Warn that DOCS_VERSION or DOCS_FRAMEWORK environment variable should be used will be thrown #9463

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

We need an extra error for performance reasons. Process should be killed when there is no environment variable `docs:start` script.

[skip changelog]

### How has this been tested?

By running `docs:start` script with/without  `DOCS_VERSION`/`DOCS_FRAMEWORK` environment variables and `docs:build` script.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9463 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ From the `docs` directory, you can run the following npm scripts:
 * `npm run docs:review [COMMIT_HASH]` – Deploys the documentation locally at a `[COMMIT_HASH]` commit.
 * `DOCS_VERSION=<semver.version> npm run docs:start:no-cache` – Starts a local documentation server, just for the <semver.version> documentation version.
 * `DOCS_VERSION=next npm run docs:start:no-cache` – Starts a local documentation server, just for the `next` documentation version.
+* `DOCS_FRAMEWORK=<javascript|react|angular|vue2|vue3> npm run docs:start:no-cache` – Starts a local documentation server, just for the framework documentation version.
```
Comment (kirszenbaum) [inline docs/README.md:63]:
```suggestion
* `DOCS_FRAMEWORK=<javascript|react|angular|vue2|vue3> npm run docs:start:no-cache` – Starts a local documentation server, just for the selected framework documentation version.
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

**Review by kirszenbaum — APPROVED** [review]

File: docs/.vuepress/config.js [hunk 2]
```diff
@@ -41,6 +42,15 @@ if (getEnvDocsVersion()) {
...
   frameworkPartialPath = `${getEnvDocsFramework()}${FRAMEWORK_SUFFIX}/`;
+
+} else if (isEnvDev()) {
+  // eslint-disable-next-line no-console
+  console.error(chalk.red(
+    'Stopping `docs:start` script execution. Any from `DOCS_FRAMEWORK` or `DOCS_FRAMEWORK` environment variables ' +
+    'should be defined for performance reason, ie. by using `DOCS_FRAMEWORK=javascript npm run docs:start:no-cache` ' +
+    'command.'
+  ));
+  process.exit(1);
```
Comment (jansiegel) [inline docs/.vuepress/config.js:53]: I'd rephrase (and reformat) that to something more along the lines of

```suggestion
  console.error(
    `${chalk.red(`\
Stopping the ${chalk.italic('docs:start')} script execution.

For performance reasons, the ${chalk.italic('DOCS_VERSION')} and/or ${chalk.italic('DOCS_FRAMEWORK')} \
environment variables need to be defined.

For example, try calling:
${chalk.italic.bold('DOCS_VERSION=next DOCS_FRAMEWORK=javascript npm run docs:start:no-cache')}.
`)}`);

  process.exit(1);
```

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: docs/.vuepress/config.js (hunk 2 above)
Comment (wszymanski) [reply docs/.vuepress/config.js]: Thank you, I've changed your suggestion just a little bit within: https://github.com/handsontable/handsontable/pull/9488/commits/7fec930d5e459482f10d6a1c8cc088a8cb238299.

**Review by jansiegel — APPROVED** [review]
